### PR TITLE
Lets the limbgrower make mobs with only a chest

### PR DESCRIPTION
**Pull request by Arturlang**
Lets medbay make custom bodies that are not originated from monkeys, still takes a good bit of effort and resources, but It's an option

## Changelog
:cl:
add: The limbgrower can make human chest parts that you can build onto into a person.
/:cl:

### CODE REVIEW

File: code/game/machinery/limbgrower.dm [hunk 1]
```diff
@@ -119,6 +126,18 @@
 	if (busy)
 		to_chat(user, "<span class=\"alert\">\The [src] is busy. Please wait for completion of previous operation.</span>")
 		return
+	if(ispath(user_item.type, /obj/item/reagent_containers/blood) && user.a_intent != INTENT_HARM)
+		var/obj/item/reagent_containers/blood/B = user_item
+		if(!B.reagents.get_reagents())
+			to_chat(user, "<span class=\"alert\">You can't fill [src] with a empty [B.name]!</span>")
```
Comment (Putnam3145) [inline code/game/machinery/limbgrower.dm]:
```suggestion
			to_chat(user, "<span class=\"alert\">You can't fill [src] with an empty [B.name]!</span>")
```

File: code/game/machinery/limbgrower.dm [hunk 2]
```diff
@@ -180,7 +199,7 @@
 			for(var/reagent_id in consumed_reagents_list)
 				consumed_reagents_list[reagent_id] *= production_coefficient
 				if(!reagents.has_reagent(reagent_id, consumed_reagents_list[reagent_id]))
-					audible_message("<span class='notice'>\The [src] buzzes.</span>")
+					audible_message("<span class='warning'>\The [src] buzzes, with a screen showing: INSUFFICENT REAGENTS</span>")
```
Comment (Putnam3145) [inline code/game/machinery/limbgrower.dm]:
```suggestion
					audible_message("<span class='warning'>\The [src] buzzes and states "INSUFFICENT REAGENTS"</span>")
```
(audible_message shows to blind characters and them being able to read the display is a bit weird)

File: code/game/machinery/limbgrower.dm [hunk 3]
```diff
@@ -227,6 +242,17 @@
 	flick("limbgrower_unfill", src)
 	icon_state = "limbgrower_idleoff"
 
+/obj/machinery/limbgrower/proc/HandleReagentUsage(modified_consumed_reagents_list, remove = TRUE)
+	for(var/reagent_id in modified_consumed_reagents_list)
+		if(!reagents.has_reagent(reagent_id, modified_consumed_reagents_list[reagent_id]))
+			audible_message("<span class='warning'>\The [src] buzzes, with a screen showing: INSUFFICENT REAGENTS</span>")
+			playsound(src, 'sound/machines/buzz-sigh.ogg', 50, FALSE)
+			break
+		if(remove)
```
Comment (Putnam3145) [inline code/game/machinery/limbgrower.dm]: move this outside of the loop, probably--much faster that way

File: code/game/machinery/limbgrower.dm [hunk 4]
```diff
@@ -306,6 +332,53 @@
 	else
 		new built_typepath(loc)
 
+/obj/machinery/limbgrower/proc/build_mob_chest(built_typepath)
+	// Create a mob with a chest, but nothing else
+	if(!ispath(built_typepath, /mob/living/carbon/human/chestonly))
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 50, FALSE)
+		visible_message(src, "Buzzes, a error screen appearing on it's display.")
```
Comment (Putnam3145) [inline code/game/machinery/limbgrower.dm]:
```suggestion
		visible_message(src, "Buzzes, an error screen appearing on its display.")
```

File: code/game/machinery/limbgrower.dm [hunk 5]
```diff
@@ -180,7 +199,7 @@
 			for(var/reagent_id in consumed_reagents_list)
 				consumed_reagents_list[reagent_id] *= production_coefficient
 				if(!reagents.has_reagent(reagent_id, consumed_reagents_list[reagent_id]))
-					audible_message("<span class='notice'>\The [src] buzzes.</span>")
+					audible_message("<span class='warning'>\The [src] buzzes and states "INSUFFICENT REAGENTS"</span>")
```
Comment (Linzolle) [inline code/game/machinery/limbgrower.dm]: please escape the double quotation marks
```suggestion
					audible_message("<span class='warning'>\The [src] buzzes and states \"INSUFFICENT REAGENTS\"</span>")
```

File: code/game/machinery/limbgrower.dm [hunk 6]
```diff
@@ -227,6 +242,25 @@
 	flick("limbgrower_unfill", src)
 	icon_state = "limbgrower_idleoff"
 
+/obj/machinery/limbgrower/proc/reagent_usage(modified_consumed_reagents_list, remove = TRUE)
+	// Apparently, having a boolean in a loop is worse than doing it twice
```
Comment (Putnam3145) [inline code/game/machinery/limbgrower.dm:246]: the loop isn't done twice, that's kinda how `else` works, but whatever

Comment (Arturlang) [reply code/game/machinery/limbgrower.dm:246]: Just meant in terms of copypasta

**Pull request merged**